src/Api/PubnubApi/EndPoint/Objects/ManageChannelMembersOperation.cs

```diff
@@ -266,20 +204,82 @@ private async Task<PNResult<PNChannelMembersResult>> ProcessMembersOperationRequ
+            var requestParameter = CreateRequestParameter();
+			Tuple<string, PNStatus> JsonAndStatusTuple;
+			var transportRequest = PubnubInstance.transportMiddleware.PreapareTransportRequest(requestParameter: requestParameter, operationType: PNOperationType.PNManageChannelMembersOperation);
+			var transportResponse = await PubnubInstance.transportMiddleware.Send(transportRequest: transportRequest);
+			if (transportResponse.Error == null) {
+				var responseString = Encoding.UTF8.GetString(transportResponse.Content);
+				PNStatus errorStatus = GetStatusIfError(requestState, responseString);
+				if (errorStatus == null) {
+					PNStatus status = new StatusBuilder(config, jsonLibrary).CreateStatusResponse(requestState.ResponseType, PNStatusCategory.PNAcknowledgmentCategory, requestState, (int)HttpStatusCode.OK, null);
+					JsonAndStatusTuple = new Tuple<string, PNStatus>(responseString, status);
+				} else {
+					JsonAndStatusTuple = new Tuple<string, PNStatus>(string.Empty, errorStatus);
+				}
+				returnValue.Status = JsonAndStatusTuple.Item2;
+            string json = JsonAndStatusTuple.Item1;
+            if (!string.IsNullOrEmpty(json))
+            {
+                List<object> resultList = ProcessJsonResponse(requestState, json);
+                ResponseBuilder responseBuilder = new ResponseBuilder(config, jsonLibrary, pubnubLog);
+                PNChannelMembersResult responseResult = responseBuilder.JsonToObject<PNChannelMembersResult>(resultList, true);
+                if (responseResult != null)
+                {
+                    returnValue.Result = responseResult;
+                }
+            }
+			} else {
+				int statusCode = PNStatusCodeHelper.GetHttpStatusCode(transportResponse.Error.Message);
+				PNStatusCategory category = PNStatusCategoryHelper.GetPNStatusCategory(statusCode, transportResponse.Error.Message);
+				PNStatus status = new StatusBuilder(config, jsonLibrary).CreateStatusResponse(PNOperationType.PNManageChannelMembersOperation, category, requestState, statusCode, new PNException(transportResponse.Error.Message, transportResponse.Error));
+				returnValue.Status = status;
+			}
+            return returnValue;
+        }
+
+        private static string MapEnumValueToEndpoint(string enumValue)
+        {
+            string ret = "";
+            if (enumValue.ToLowerInvariant() == "custom")
+            {
+                ret = "custom";
+            }
+            else if (enumValue.ToLowerInvariant() == "uuid")
+            {
+                ret = "uuid";
+            }
+            else if (enumValue.ToLowerInvariant() == "channel")
+            {
+                ret = "channel";
+            }
+            else if (enumValue.ToLowerInvariant() == "channel_custom")
+            {
+                ret = "channel.custom";
+            }
+            else if (enumValue.ToLowerInvariant() == "uuid_custom")
+            {
+                ret = "uuid.custom";
```

Comment:
I think I've seen this conversion in another file, can this method be put once into a util?

Reply:
It's required conversation between C# request object and Rest api query parameter key values, So if something is going to change, we need to update this conversation.

Reply:
I mean if we can move MapEnumValueToEndpoint method to a static util since it's repeated at endpoints but doesn't seem to have significant variation in how it translates the values?

Reply:
Got it.
Agree! moving it to a common place...

Reply:
Refactored! at [bbd62a4](https://github.com/pubnub/c-sharp/pull/225/commits/bbd62a40868a47ffe21a60830ae97e93f67ab290)

good catch!